---
layout: cv
title: James Collier Résumé
---
# Isaac Newton
Physicist, Mathematician, Cambridge professor.

<div id="webaddress">
<a href="james@thecolliers.xyz">james@thecolliers.xyz</a>
</div>


## Interests

Reliable systems, functional programming, software engineering, scientific computing, open-source software.

## Experience

`2019-now`
__Flemish Institute of Biotechnology, Gent, Belgium__
Bioinformatics Tool Developer

`2019`
__Topic Healthcare Solutions, Best, The Netherlands__
Embedded Software Engineer

`2018`
__IMec Connected Health Solutions, Eindhoven, The Netherlands__

`2017-2018`
__Topic Embedded Systems, Best, The Netherlands__
Embedded Software Engineer

`2012-2015`
__Monash University, Clayton, Australia__


## Education

`2012-2016`
__Doctor of Philosophy, Monash University, Clayton, Australia__

`2011`
__Honours Degree of the Bachelor of Computer Science, Monash University, Clayton, Australia__

`2008-2010`
__Bachelor of Science, Monash University, Clayton, Australia__

## Technical Experience

| Language   | Tool      | Framework  |
|------------|-----------|------------|
| Purescript | Spago     | React      |
| Elm        | Purs      | Elm        |
| Haskell    | Purs-Tidy | httpurple  |
| Rust       | Zephyr    | FastAPI    |
| Javascript | Cabal     | SQLAlchemy |
| Typescript | Emacs     | PyTest     |
| SQL        | NPM       | Concur     |
| C          | Nix       | Redux      |
| Python     | Node.js   | Redux-Saga |

## Software

* [Zephyr](https://github.com/MaybeJustJames/zephyr) Dead code elimination tool for [Purescript](https://purescript.org/) written in [Haskell](https://haskell.org/).
* [yaml](https://github.com/MaybeJustJames/yaml) YAML encoding/decoding library for [Elm](https://elm-lang.org/).
* [Phylio](https://github.com/vibbits/phylio) [Purescript](https://purescript.org) library for parsing phylogenetic trees.
* [VIB-CSS](https://github.com/vibbits/vib-css) CSS library for styling VIB websites and apps.
* [Super](https://gitlab.com/structural-fragment-search/super) 3D pattern matching on proteins.
* [SCope](https://github.com/aertslab/SCope) web app to visualise single-cell transcriptomics data.
* [MMLigner](https://lcb.infotech.monash.edu/mmligner/) align protein structures based on [Minimum Message Length](http://allisons.org/ll/MML/) inference.

## Awards

`2017`
_Vice-Chancellors Commendation For PhD Thesis Excellence_ by Monash University.

`2016`
Recommended for _"Best Thesis"_ award by external thesis examiner, Prof. Anna Tramontano.

`2013`
Recipient of _Victorian Life Sciences Supercomputing Initiative_ (VLSCI) Travel Grant.

`2012-2016`
Recipient of _Australian Postgraduate Award_ (APA) Postgraduate scholarship.

`2012-2016`
Recipient of _National ICT Australia_ (NICTA) Postgraduate scholarship.

`2009`
Received an _Undergraduate Vacation Scholarship_.

`2008`
Membership to the _Golden Key International Honour Society_ for outstanding achievement.


## Publications

`In correspondence`
**Acinetobase: the comprehensive database and repository of Acinetobacter strains.**

`2022-01-03 in Bioinformatics`
**Unipept Visualizations: an interactive visualization library for biological data**
[10.1093/bioinformatics/btab590](https://doi.org/10.1093/bioinformatics/btab590)

`2021-05-17 in Nature Communications`
**AutoSpill is a principled framework that simplifies the analysis of multichromatic flow cytometry data**
[10.1038/s41467-021-23126-8](https://doi.org/10.1038/s41467-021-23126-8)

`2017-01-08 in Bioinformatics`
**Statistical inference of protein structural alignments using information and compression**
[10.1093/bioinformatics/btw757](https://doi.org/10.1093/bioinformatics/btw757)

`2016-06-02 in Journal of Computational Biology`
**On Sufficient Statistics of Least-Squares Superposition of Vector Sets**
[10.1089/cmb.2014.0154](https://doi.org/10.1089/cmb.2014.0154)

`2014-08 in Bioinformatics`
**A new statistical framework to assess structural alignment quality using information compression**
[10.1093/bioinformatics/btu460](https://doi.org/10.1093/bioinformatics/btu460)

`2012-05 in Nucleic Acids Research`
**Super: a web server to rapidly screen superposable oligopeptide fragments from the protein data bank**
[10.1093/nar/gks436](https://doi.org/10.1093/nar/gks436)


