---
layout: cv
title: James Collier Résumé
---
# Dr. James Collier
Software Engineer.

<div id="webaddress">
    <span>
        <a href="james@thecolliers.xyz">james@thecolliers.xyz</a>
    </span>
    <span>
        <img alt="Twitter" src="https://img.shields.io/twitter/url?label=Twitter&style=social&url=https%3A%2F%2Ftwitter.com%2FMaybeJustJames">
    </span>
    <span>
        <img alt="GitHub User's stars" src="https://img.shields.io/github/stars/MaybeJustJames?label=GitHub&style=social">
    </span>
    <span>
        <img alt="GitLab stars" src="https://img.shields.io/gitlab/stars/structural-fragment-search/super?label=GitLab&style=social">
    </span>
</div>


## Interests

Reliable systems, functional programming, software engineering, scientific computing, open-source software.

## Experience

10 years of professional software engineering.

`2019-current`
__Flemish Institute of Biotechnology, Gent, Belgium__
Bioinformatics Tool Developer

`2019`
__Topic Healthcare Solutions, Best, The Netherlands__
Embedded Software Engineer

`2018`
__IMec Connected Health Solutions, Eindhoven, The Netherlands__
Embedded Software Engineer

`2017-2018`
__Topic Embedded Systems, Best, The Netherlands__
Embedded Software Engineer

`2012-2015`
__Monash University, Clayton, Australia__
Software Engineer


## Education

`2012-2016`
__Doctor of Philosophy, Monash University, Clayton, Australia ([Thesis](https://doi.org/10.4225/03/58b79813d9110))__

`2011`
__Honours Degree of the Bachelor of Computer Science, Monash University, Clayton, Australia__

`2008-2010`
__Bachelor of Science, Monash University, Clayton, Australia__

## Technical Experience

`Languages`
**[Purescript](https://www.purescript.org)**, **[Elm](https://elm-lang.org)**, **[Haskell](https://www.haskell.org)**, **[Rust](https://www.rust-lang.org)**, **[Javascript](http://www.jsfuck.com/)**, **[Typescript](https://www.typescriptlang.org)**, **[SQL](https://www.iso.org/standard/63555.html)**, **[C](https://www.open-std.org/jtc1/sc22/wg14/)**, **[Python](https://www.python.org)**, **[Emacs Lisp](https://www.gnu.org/software/emacs)**, **[Common Lisp](http://clhs.lisp.se)**, **[C++](https://isocpp.org)**, **[Java](https://openjdk.org)**

`Tools`
**[Emacs](https://www.gnu.org/software/emacs)**, **[Spago](https://github.com/purescript/spago)**, **[Cargo](https://doc.rust-lang.org/cargo)**, **[Cabal](https://www.haskell.org/cabal)**, **[Nix](https://nixos.org)**, **[Poetry](https://python-poetry.org)**, **[npm](https://www.npmjs.com)**

`Frameworks | Libraries`
**[React](https://reactjs.org)**, **[FastAPI](https://fastapi.tiangolo.com)**, **[PyTest](https://docs.pytest.org)**, **[Hadgehog](https://hedgehog.qa)**, **[Hypothesis](https://hypothesis.works)**, **[Redux](https://redux.js.org)**, **[Servant](https://www.servant.dev)**, **[Rel8](https://rel8.readthedocs.io)**, **[SQLAlchemy](https://www.sqlalchemy.org)**, 

## Software

* [Zephyr](https://github.com/MaybeJustJames/zephyr): Dead code elimination tool for [Purescript](https://purescript.org/) written in [Haskell](https://haskell.org/).
* [yaml](https://github.com/MaybeJustJames/yaml): YAML encoding/decoding library for [Elm](https://elm-lang.org/).
* [Phylio](https://github.com/vibbits/phylio): [Purescript](https://purescript.org) library for parsing phylogenetic trees.
* [VIB-CSS](https://github.com/vibbits/vib-css): CSS library for styling websites and apps for the Flemish Institute for Biotechnology.
* [Super](https://gitlab.com/structural-fragment-search/super): 3D pattern matching on proteins written in C, also a [Python wrapper library](https://pypi.org/project/pysuper/).
* [SCope](https://github.com/aertslab/SCope): web app to visualise single-cell transcriptomics data. Backend written in [Python](https://python.org/) using [FastAPI](https://fastapi.tiangolo.com/) and [SQLAlchemy](https://www.sqlalchemy.org/). Frontend written in [Typescript](https://www.typescriptlang.org/) using [React](https://reactjs.org/), [Redux](https://redux.js.org/), and [Redux-Saga](https://redux-saga.js.org/).
* [Falcon-Client](https://bitbucket.org/kloostermannerflab/falcon-swr-client): GUI Client application for [Falcon](https://bitbucket.org/kloostermannerflab/falcon-core) closed-loop neuroscience platform. Uses [Rust](https://www.rust-lang.org/) to communicate with Falcon over [ZeroMQ](https://zeromq.org/). Uses [Elm](https://elm-lang.org/) to display a web GUI interface with real time plotting.
* [MMLigner](https://lcb.infotech.monash.edu/mmligner/): align protein structures based on [Minimum Message Length](http://allisons.org/ll/MML/) inference. Written in C++.
* [Golden Gate Assembler](https://github.com/vibbits/GGW-Elm): Store and organise genetic products for robotic vector assembly. Simulate genetic vector assembly. Written in Elm and Python.
* [Acinetobase](https://github.com/vibbits/acinetobase-static): Static site generated from data in a CSV table and images. Generator written in Haskell.

## Awards

`2017`
_Vice-Chancellors Commendation For PhD Thesis Excellence_ by Monash University.

`2016`
Recommended for _"Best Thesis"_ award by external thesis examiner, Prof. Anna Tramontano.

`2013`
Recipient of _Victorian Life Sciences Supercomputing Initiative_ (VLSCI) Travel Grant.

`2012-2016`
Recipient of _Australian Postgraduate Award_ (APA) Postgraduate scholarship.

`2012-2016`
Recipient of _National ICT Australia_ (NICTA) Postgraduate scholarship.

`2009`
Received an _Undergraduate Vacation Scholarship_.

`2008`
Membership to the _Golden Key International Honour Society_ for outstanding achievement.


## Publications

`In correspondence`
**Acinetobase: the comprehensive database and repository of Acinetobacter strains.**

`2022-01-03 in Bioinformatics`
**Unipept Visualizations: an interactive visualization library for biological data**
[10.1093/bioinformatics/btab590](https://doi.org/10.1093/bioinformatics/btab590)

`2021-05-17 in Nature Communications`
**AutoSpill is a principled framework that simplifies the analysis of multichromatic flow cytometry data**
[10.1038/s41467-021-23126-8](https://doi.org/10.1038/s41467-021-23126-8)

`2017-01-08 in Bioinformatics`
**Statistical inference of protein structural alignments using information and compression**
[10.1093/bioinformatics/btw757](https://doi.org/10.1093/bioinformatics/btw757)

`2016-06-02 in Journal of Computational Biology`
**On Sufficient Statistics of Least-Squares Superposition of Vector Sets**
[10.1089/cmb.2014.0154](https://doi.org/10.1089/cmb.2014.0154)

`2014-08 in Bioinformatics`
**A new statistical framework to assess structural alignment quality using information compression**
[10.1093/bioinformatics/btu460](https://doi.org/10.1093/bioinformatics/btu460)

`2012-05 in Nucleic Acids Research`
**Super: a web server to rapidly screen superposable oligopeptide fragments from the protein data bank**
[10.1093/nar/gks436](https://doi.org/10.1093/nar/gks436)


## Presentations &amp; Talks

`VIB course: Gentle Hands-On Introduction to Python Programming (2022)`
**How to setup a development environment for Python**
[YouTube](https://youtu.be/vLRAUHpeHtM)

`Internal presentation`
**Background task execution**
[Slides](https://maybejustjames.github.io/background-tasks-talk)

`European Conference on Computational Biology (2014)`
**A new statistical framework to assess structural alignment quality using information compression**
[Slides](https://www.dropbox.com/s/4h3itvwwflvwl0t/Mon7_James_Collier.pdf?dl=0)

`Three Minute Thesis (2013)`
**Protein Structural Alignment Problem in Biology**
[YouTube](https://youtu.be/h0BY3lcEFKQ)
